### PR TITLE
[prometheus-fastly-exporter] Add fastly exporter

### PR DESCRIPTION
**Pull request by arslanbekov**
Added new exporter for https://fastly.com service
https://github.com/fastly/fastly-exporter

### CODE REVIEW

**Comment by arslanbekov** [comment]
Hey `@helm-charts-admins`
1. Who should I specify in `prometheus-fastly-exporter/Chart.yaml` to pass the check `Check CODEOWNERS`?
2. How to pass the check by README.md the word `repo` is checked there (there is in `helm repo ...`)?

File: charts/prometheus-fastly-exporter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+appVersion: "7.2.4"
+description: A Helm chart for the Prometheus Fastly Exporter
+name: prometheus-fastly-exporter
+version: 1.0.0
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/Chart.yaml:5]: First version should start at 0.1.0

File: charts/prometheus-fastly-exporter/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+
+## Install Chart
+
+```console
+# Helm 3
+$ helm install [RELEASE_NAME] prometheus-community/prometheus-fastly-exporter
+
+# Helm 2
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/README.md:26]: Helm 2 is deprecated. Please remove all references to it.

**Review by monotek — CHANGES_REQUESTED** [review]
Please fix linting errors and ad yourself to the code owners file.

**Comment by arslanbekov** [comment]
@monotek could you merge it please?

**Comment by monotek** [comment]
no, please fix the errors first.
just check readme of other charts.

**Comment by arslanbekov** [comment]
@monotek Are you sure that this check works validly? I can't change the link and command. In other README.md it is present (but maybe ignored?).



Screenshot 2022-07-21 at 15 35 35

File: charts/prometheus-fastly-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+- Kubernetes 1.8+ with Beta APIs enabled
+
+## Get Repository Info
+
+```console
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+```
+
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/README.md:17]:
```suggestion

_See [`helm repo`](https://helm.sh/docs/helm/helm_repo/) for command documentation._
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by arslanbekov** [comment]
@monotek thank you, all sorted now?

**Comment by monotek** [comment]
No, the chart itsef fails to start the container: https://github.com/prometheus-community/helm-charts/runs/7448786751?check_suite_focus=true

File: charts/prometheus-fastly-exporter/values.yaml [hunk 4]
```diff
@@ -21,7 +21,8 @@ image:
   tag: v7.2.4
   pullPolicy: IfNotPresent
 
-token: ""
+# API token Fastly - https://docs.fastly.com/en/guides/using-api-tokens#creating-api-tokens
+token: "token"
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/values.yaml:25]: It shoudl also be possible to use the token from an existing secret.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-fastly-exporter/values.yaml (hunk 4 above)
Comment (arslanbekov) [reply charts/prometheus-fastly-exporter/values.yaml:25]: Added new `envTokenFromSecret` variable

File: charts/prometheus-fastly-exporter/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            {{- if not .Values.envTokenFromSecret }}
+            - "-token={{ .Values.token }}"
+            {{- end }}
+            {{- range $key, $value := .Values.options }}
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/templates/deployment.yaml:38]: why is this in a loop?
can there be multiple tokens?

can we do it similar to: https://github.com/prometheus-community/helm-charts/blob/5ce650b75c3bb4cc85e0f7c23ac2614bb37d67de/charts/prometheus-mongodb-exporter/templates/deployment.yaml#L31-L35

So the token is always stored and referenced from a secret and also the key in the secret is configurable?

More common naming for the variable would be "existingSecret".

Comment (arslanbekov) [reply charts/prometheus-fastly-exporter/templates/deployment.yaml:38]: Thanks for the ref. 

The token is not in a loop.
The loop contains keys that can be passed like this: https://github.com/fastly/fastly-exporter#filtering-services

Comment (arslanbekov) [reply charts/prometheus-fastly-exporter/templates/deployment.yaml:35]: And no, there must be only one token.

I slightly modified the definition of the variable.
Now it is always set from env: `FASTLY_API_TOKEN`

**Comment by arslanbekov** [comment]
@monotek I used [prometheus-elasticsearch-exporter](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-elasticsearch-exporter) as a template for this chart. 
But as I understand it, it is a bit outdated.

I fixed the use of token from `existSecret` and the use of the new common labels.

**Comment by arslanbekov** [comment]
[@monotek](https://github.com/monotek) I used [prometheus-elasticsearch-exporter](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-elasticsearch-exporter) as a template for this chart.
But as I understand it, it is a bit outdated.

Also fixed all your suggestions.

**Comment by arslanbekov** [comment]
@mrueg hello, can review this, please?

File: charts/prometheus-fastly-exporter/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
```
Comment (monotek) [inline charts/prometheus-fastly-exporter/Chart.yaml:1]:
```suggestion
apiVersion: v2
```

**Review by monotek — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Please update chart API version

**Review by monotek — APPROVED** [review]

**Pull request merged**